File: app/src/main/java/wangdaye/com/geometricweather/daily/adapter/holder/UVHolder.java
package wangdaye.com.geometricweather.daily.adapter.holder;

import android.content.Context;
import android.content.res.ColorStateList;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.appcompat.widget.AppCompatImageView;
import androidx.core.widget.ImageViewCompat;

import wangdaye.com.geometricweather.R;
import wangdaye.com.geometricweather.basic.models.weather.UV;
import wangdaye.com.geometricweather.daily.adapter.DailyWeatherAdapter;
import wangdaye.com.geometricweather.daily.adapter.model.DailyUV;

public class UVHolder extends DailyWeatherAdapter.ViewHolder {

    private final AppCompatImageView mIcon;
    private final TextView mTitle;

    public UVHolder(ViewGroup parent) {
        super(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_weather_daily_uv, parent, false));
        mIcon = itemView.findViewById(R.id.item_weather_daily_uv_icon);
        mTitle = itemView.findViewById(R.id.item_weather_daily_uv_title);
    }

    @Override
    public void onBindView(DailyWeatherAdapter.ViewModel model, int position) {
        Context context = itemView.getContext();
        UV uv = ((DailyUV) model).getUv();

        ImageViewCompat.setImageTintList(mIcon, ColorStateList.valueOf(uv.getUVColor(context)));
        mTitle.setText(uv.getUVDescription());

        itemView.setContentDescription(context.getString(R.string.uv_index) + ", " + mTitle.getText());
    }
}

File: app/src/main/java/org/breezyweather/daily/adapter/holder/UVHolder.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.daily.adapter.holder

import android.content.res.ColorStateList
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.widget.AppCompatImageView
import androidx.core.widget.ImageViewCompat
import org.breezyweather.R
import org.breezyweather.daily.adapter.DailyWeatherAdapter
import org.breezyweather.daily.adapter.model.DailyUV
import org.breezyweather.domain.weather.model.getUVColor
import org.breezyweather.domain.weather.model.getUVDescription

class UVHolder(parent: ViewGroup) : DailyWeatherAdapter.ViewHolder(
    LayoutInflater.from(parent.context).inflate(R.layout.item_weather_daily_uv, parent, false)
) {
    private val mIcon: AppCompatImageView = itemView.findViewById(R.id.item_weather_daily_uv_icon)
    private val mTitle: TextView = itemView.findViewById(R.id.item_weather_daily_uv_title)

    override fun onBindView(model: DailyWeatherAdapter.ViewModel, position: Int) {
        val context = itemView.context
        val uv = (model as DailyUV).uv
        ImageViewCompat.setImageTintList(
            mIcon,
            ColorStateList.valueOf(uv.getUVColor(context))
        )
        mTitle.text = uv.getUVDescription(context)
        itemView.contentDescription = context.getString(R.string.uv_index) +
            context.getString(R.string.comma_separator) + mTitle.text
    }
}
